pyomo/core/base/constraint.py

```diff
@@ -641,6 +613,12 @@ def set_value(self, expr):
                     "non-finite term." % (self.name))
             assert self._lower is self._upper
 
+        #
+        # Compress the body expression
+        #
+        self._body = EXPR.compress_expression(self._body)
```

Comment:
What is compress expression, why do we need it, and even if we need it, why should it be done here?  I think this should be deleted.

Reply:
Agreed.  This function is no longer needed.